src/native/eventpipe/ds-dump-protocol.c

```diff
@@ -93,6 +93,71 @@ dump_protocol_helper_unknown_command (
 	return true;
 }
 
+static
+bool
+dump_protocol_generate_core_dump_response_flatten (
+	void *payload,
+	uint8_t **buffer,
+	uint16_t *size)
+{
+    DiagnosticsGenerateCoreDumpResponsePayload *response = (DiagnosticsGenerateCoreDumpResponsePayload*)payload;
+
+	EP_ASSERT (payload != NULL);
+	EP_ASSERT (buffer != NULL);
+	EP_ASSERT (*buffer != NULL);
+	EP_ASSERT (size != NULL);
+
+	bool success = true;
+
+	// int32_t error
+	memcpy (*buffer, &response->error, sizeof (response->error));
+	*buffer += sizeof (response->error);
+	*size -= sizeof (response->error);
+
+	// LPCWSTR error message
+	success &= ds_ipc_message_try_write_string_utf16_t (buffer, size, response->error_message);
+
+	// Assert we've used the whole buffer we were given
+	EP_ASSERT(*size == 0);
+
+    return success;
+}
+
+static
+void
+dump_protocol_generate_core_dump_response (
+	DiagnosticsIpcStream *stream,
+	ds_ipc_result_t error,
+    ep_char8_t * errorText)
+{
+    DiagnosticsGenerateCoreDumpResponsePayload payload;
+	DiagnosticsIpcMessage message;
+	ds_ipc_message_init (&message);
+
+    // Initialize payload
+    payload.error = error;
+    payload.error_message = ep_rt_utf8_to_utf16_string (errorText, -1);
+
+    // Get the size of the payload
+    size_t payload_size = sizeof(payload.error);
```

Comment:
Other payloads implement this as a xxxx_payload_get_size that returns a uint16_t, of course not needed, but would make it more maintainable if/when we need to calculate the size of the payload in other places.

Reply:
I think putting it in a `uint16_t ..._get_size` function will make maintenance easier in the long term. Calculating the size can be defined next to the struct definition in the .h file and not somewhere in the .c file.